iroh/src/magicsock/relay_actor.rs

```diff
@@ -274,9 +273,24 @@ impl ActiveRelayActor {
     async fn run(mut self) -> anyhow::Result<()> {
         inc!(MagicsockMetrics, num_relay_conns_added);
 
-        loop {
-            let Some(client) = self.run_dialing().instrument(info_span!("dialing")).await else {
-                break;
+        let mut backoff = ExponentialBuilder::new()
+            .with_min_delay(Duration::from_millis(10))
+            .with_max_delay(Duration::from_secs(5))
+            .without_max_times()
+            .build();
+
+        'run: loop {
+            let client = 'dial: loop {
```

Comment:
Why does this need two loops now?

Could we not have an inner "dial_and_connect" function that does `run_dialing()?; run_connected()?` (ish) as a single shot.  Then we only have one place were we need to sleep which seems simpler.

Reply:
True, refactored this into a single loop.